openwisp_radius/tests/test_api.py

```diff
@@ -2414,3 +2414,64 @@ def test_change_phone_number_restriction(self):
             self.assertEqual(user.phone_number, phone_number)
 
         app_settings.ALLOWED_MOBILE_PREFIXES = []
+
+
+class TestApiRejectPhoneToken(ApiTokenMixin, BaseTestCase):
+
+    _test_email = 'test@openwisp.org'
+
+    def setUp(self):
+        super().setUp()
+        radius_settings = self.default_org.radius_settings
+        radius_settings.sms_verification = False
+        radius_settings.sms_sender = '+595972157632'
+        radius_settings.sms_meta_data = {
+            'clientId': 3,
+            'clientSecret': 'p8pVUoqz7T4ArPEu1rcgzlsYJmnGCkkWHs2WAqqm',
+        }
+        radius_settings.full_clean()
+        radius_settings.save()
+
+    def test_register_201(self):
+        self._superuser_login()
+        self.assertEqual(User.objects.count(), 1)
+        url = reverse('radius:rest_register', args=[self.default_org.slug])
+        phone_number = '+393664255801'
+        r = self.client.post(
+            url,
+            {
+                'username': self._test_email,
+                'email': self._test_email,
+                'password1': 'password',
+                'password2': 'password',
+                'phone_number': phone_number,
+            },
+        )
+        self.assertEqual(r.status_code, 201)
+        self.assertIn('key', r.data)
+        self.assertEqual(User.objects.count(), 2)
+        user = User.objects.get(email=self._test_email)
+        self.assertTrue(user.is_member(self.default_org))
+        self.assertEqual(user.phone_number, phone_number)
+        self.assertTrue(user.is_active)
+
+    def test_create_phone_token_403(self):
+        url = reverse('radius:phone_token_create', args=[self.default_org.slug])
+        r = self.client.post(url)
+        self.assertEqual(r.status_code, 403)
```

Comment:
would be good to check also the error message to be 100% sure, eg: `self.assertIn('SMS verification', str(r.data))`, applies for the other tests as well.